CYOAfeb21

```diff
@@ -0,0 +1,225 @@
+###Setters and Getters for Battlers###
+#Attack	
+	def attack(self):
+		return self.attack
+	def attack(self, value):
+		self.value= value
+#Defense	
+	def defense(self):
+		return self.defense
+	def defense(self, value):
+		self.value= value
+#Health
+	def health(self):
+		return self.health
+	def health(self, value):
+		self.value= value	
+#Magic
+	def magic(self):
+		return self.magic
+	def magic(self, value):
+		self.value= value	
+#Experience	
+	def experience(self):
+		return self.experience
+	def experience(self, value):
+		self.value= value
+#Level
+	def level(self):
+		return level.health
+	def level(self, value):
+		self.value= value	
+#Playclass
+	def playclass(self):
+		return self.playclass
+	def playclass(self, value):
+		self.value= value
+		
+###Start PC 'player'.###
+player= Battler(givename(), "Town", None, 1, 1, 1, 'none', 0, 1, giveclass())
+if player.playclass== 'A':
+	player.playclass='Warrior'
+	player.items= 'Sword'
+	player.attack= 5
+	player.defense= 10
+	player.health= 15
+	player.magic= 'None'
+elif player.playclass=='B':
+	player.playclass='Wizard'
+	player.items='Wand'
+	player.attack= 10
+	player.defense= 5
+	player.health= 10
+	player.magic= 'Fire'
+elif player.playclass=='C':
+	player.playclass='Cleric'
+	player.items='Relic'
+	player.attack= 4
+	player.defense= 10
+	player.health= 12
+	player.magic= 'Heal'
+elif player.playclass=='D':
+	player.playclass='Rogue'
+	player.items='Dagger' +' ' + 'Lockpick'
+	player.attack= 8
+	player.defense= 8
+	player.health= 10
+	player.magic= 'None'
+else:
+		None
+
+#Function to level up player#
+def levelup():
+		check=player.level
+		if player.experience<5:
+			player.level= 1
+		elif player.experience<10:
+			player.level= 2
+		elif player.experience<20:
+			player.level= 3
+		elif player.experience>20:
+			player.level= 4
+		else:
+			None
+		if player.level != check:
+			print("You leveled up!")
+		else:
+			None
+		
+#Function to call player statblock
+def playerstatblock():
+	self= player
+	statblock=print(f"""
+	Name: {self.name}
+	Position: {self.position}
+	Items: {self.items}
+	Attack: {self.attack}
+	Defense: {self.defense}
+	Health: {self.health}
+	Magic: {self.magic}
+	Experience: {self.experience}
+	Level: {self.level}
+	Class: {self.playclass}""")
+	return self
+playerstatblock()
+
+###Enemies given classes that will later be associated with different outcomes based on what stats the player has.###
+
+###Initiate Battler minotaur###
+minotaur= Battler('Minotaur', "Dungeon", 'Health Potion', 12, 6, 20, 'None', 5, 1, 'Beast')
+def minotaurstatblock():
+	self= minotaur
+	statblock=print(f"""
+	Name: {self.name}
+	Position: {self.position}
+	Items: {self.items}
+	Attack: {self.attack}
+	Defense: {self.defense}
+	Health: {self.health}
+	Magic: {self.magic}
+	Experience: {self.experience}
+	Level: {self.level}
+	Class: {self.playclass}""")
+#minotaurstatblock()
+
+###Function for performing a battle with the minotaur.###
+def minotaurbattle():
```

Comment:
```
def battle(attacker, defender):
  if logic_deciding_if_attack_succeeds:
    attacker.items = attacker.items + " " + defender.items
    attacker.experience = attacker.experience + defender.experience
    # Only apply damage if you need to
    if attacker.attack > defender.defense:
      defender.health = defender.health - (attacker.attack + defender.defense)
    if player.playclass== 'Wizard':
      print(f"Your {attacker.magic} magic defeats the {defender.name}. You gain {defender.experience} experience.")
    else player.playclass!='Wizard':
      if defender.attack - attacker.defense > 0:
        print(f"You fight the {defender.name} and take {defender.attack - attacker.defense} damage.")
	player.health= player.health - (minotaur.attack - player.defense)


print("Your minotaur stuff")

Reply:
Generalized battle schematic. Needs a lot of elif added.
Wiz>Beast
Warrior>Human
Cleric>Undead
Rogue>Drake